File: xssfinder-java-executor/src/main/java/org/xssfinder/scanner/MethodDefinitionFactory.java
package org.xssfinder.scanner;

import org.xssfinder.SubmitAction;
import org.xssfinder.TraverseWith;
import org.xssfinder.remote.MethodDefinition;
import org.xssfinder.remote.PageDefinition;

import java.lang.reflect.Method;

public class MethodDefinitionFactory {
    public MethodDefinition createMethodDefinition(Method method) {
        String identifier = method.getName();
        Class<?> returnTypeClass = method.getReturnType();
        Class<?> owningTypeClass = method.getDeclaringClass();
        boolean hasArgs = method.getParameterTypes().length > 0;
        boolean isSubmit = method.isAnnotationPresent(SubmitAction.class);
        boolean hasCustomTraverser = method.isAnnotationPresent(TraverseWith.class);
        return new MethodDefinition(
                identifier,
                returnTypeClass.getCanonicalName(),
                owningTypeClass.getCanonicalName(),
                hasArgs,
                isSubmit,
                hasCustomTraverser
        );
    }
}

File: xssfinder-php-executor/src/XssFinder/Scanner/MethodDefinitionFactory.php
<?php

namespace XssFinder\Scanner;

use XssFinder\MethodDefinition;

class MethodDefinitionFactory
{
    /** @var \XssFinder\Scanner\ReflectionHelper */
    private $_reflectionHelper;

    public function __construct(ReflectionHelper $reflectionHelper)
    {
        $this->_reflectionHelper = $reflectionHelper;
    }

    /**
     * @param $method \ReflectionMethod
     * @return MethodDefinition
     */
    public function createMethodDefinition(\ReflectionMethod $method)
    {
        $methodDefinition = new MethodDefinition();
        $methodDefinition->identifier = $method->getName();
        $methodDefinition->returnTypeIdentifier = $this->_reflectionHelper->getReturnType($method);
        $methodDefinition->owningTypeIdentifier = '\\' . $method->getDeclaringClass()->getName();
        $methodDefinition->submitAnnotated = $this->_reflectionHelper->isSubmitAnnotated($method);
        $methodDefinition->customTraversed = $this->_reflectionHelper->isTraverseWithAnnotated($method);
        $methodDefinition->parameterised = $method->getNumberOfRequiredParameters() > 0;
        return $methodDefinition;
    }
}